File: puccini/examples/ruby/compile.rb
#!/usr/bin/env ruby

require 'puccini'
require 'yaml'

if ARGV.length == 0
  puts 'no URL provided'
  exit 1
end

begin
  clout = Puccini::TOSCA.compile(ARGV[0])
  puts YAML.dump(clout)
rescue Puccini::TOSCA::Problems => e
  puts 'Problems:'
  for problem in e.problems
    puts YAML.dump(problem)
  end
  exit 1
end

File: puccini/examples/python/compile.py
#!/usr/bin/env python3

# Note that installing `puccini` will also install `ard` 

import sys, puccini.tosca, ard

if len(sys.argv) <= 1:
    sys.stderr.write('no URL provided\n')
    sys.exit(1)

url = sys.argv[1]

try:
    clout = puccini.tosca.compile(url)
    ard.write(clout, sys.stdout)
except puccini.tosca.Problems as e:
    print('Problems:', file=sys.stderr)
    for problem in e.problems:
        ard.write(problem, sys.stderr)
    sys.exit(1)
